lib/user-interface/react/src/App.tsx

```diff
@@ -101,80 +101,90 @@ function App () {
-                footerSelector='#f'
-                navigationHide={!showNavigation}
-                breadcrumbs={<Breadcrumbs />}
-                toolsHide={true}
-                notifications={<NotificationBanner />}
-                stickyNotifications={true}
-                navigation={nav}
-                navigationWidth={450}
-                content={
-                    <Routes>
-                        <Route index path='*' element={<Home setNav={setNav} />} />
-                        <Route
-                            path='chatbot'
-                            element={
-                                <PrivateRoute>
-                                    <Chatbot setNav={setNav} />
-                                </PrivateRoute>
-                            }
-                        />
-                        <Route
-                            path='chatbot/:sessionId'
-                            element={
-                                <PrivateRoute>
-                                    <Chatbot setNav={setNav} />
-                                </PrivateRoute>
-                            }
-                        />
-                        <Route
-                            path='model-management'
-                            element={
-                                <AdminRoute>
-                                    <ModelManagement setNav={setNav} />
-                                </AdminRoute>
-                            }
-                        />
-                        <Route
-                            path='document-library'
-                            element={
-                                <PrivateRoute showConfig='showRagLibrary' configs={config}>
-                                    <RepositoryLibrary setNav={setNav} />
-                                </PrivateRoute>
-                            }
-                        />
-                        <Route
-                            path='document-library/:repoId'
-                            element={
-                                <PrivateRoute showConfig='showRagLibrary' configs={config}>
-                                    <DocumentLibrary setNav={setNav} />
-                                </PrivateRoute>
-                            }
-                        />
-                        <Route
-                            path='configuration'
-                            element={
-                                <AdminRoute>
-                                    <Configuration setNav={setNav} />
-                                </AdminRoute>
-                            }
-                        />
-                    </Routes>
-                }
-            />
-            {confirmationModal && <ConfirmationModal {...confirmationModal} />}
-            {config?.configuration.systemBanner.isEnabled && <SystemBanner position='BOTTOM' />}
+            <ConfigurationContext.Provider value={config}>
```

Comment:
I updated this to create a provider for our whole app. Instead of each component that needs the config loading it themselves.